javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,30 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+
+const filter = document.getElementById("filter");
+let h3 = document.getElementsByName("h3");
```

Comment:
I don’t get it, can you explain more about hyfBay

Reply:
Yes, it's what Javascript does when you do get .getElementByName (which would be searching for a fit in the "name" attribute of your element) or .getElementByTagName( h3 is  tag name, the same as "div", "p", or other html stuff). HTMLCollection is an array like object, but it doesnt have the methods (try console log what you have now, it looks like an array but when you click on the little arrow it says it's HTMLCollection and it doesnt have the same properties as an array)..similarly as when you search for things with .querySelector("h3") , it would return a NodeList that is not really an array so you can not call all the array methods on it (such as .map, .filter, .every ..)

Reply:
Here is an article on difference between NodeLists and HTMLCollections , which are both array-like objects but not arrays ..so one can not do the array magic :( https://www.w3schools.com/js/js_htmldom_nodelist.asp

Reply:
I am sorry if I am not explaining it too well. It's not too intuitive :/ since it really looks like an array

Reply:
Fixed